docs/docs/35-references/10-promotion-steps.md

```diff
@@ -5,10 +5,93 @@ description: Learn about all of Kargo's built-in promotion steps
 
 # Promotion Steps Reference
 
-This reference document describes the promotion steps that are built directly
-into Kargo. Steps are presented roughly in the order in which they might appear
-in a typical promotion process. Similarly, configuration options for each step
-are laid out in order of their applicability to typical use cases.
+## Introduction
+
+Kargo's promotion steps are the building blocks of a promotion process. They
+perform the actions necessary to promote a piece of Freight into a Stage.
+Promotion steps are designed to be composable, allowing users to construct
+complex promotion processes from simple, reusable components.
+
+### Defining a Promotion Step
+
+A promotion step is a YAML object with at least one key, `uses`, whose value is
+the name of the step to be executed. The step's configuration is provided in a
+subsequent key, `config`. The `config` key's value is an object whose keys are
+the configuration options for the step.
+
+```yaml
+steps:
+- uses: step-name
+  config:
+    option1: value1
+    option2: value2
+```
+
+:::info
+For a list of built-in promotion steps and configuration options, see the
+[Built-in Steps](#built-in-steps) section.
+:::
+
+#### Step Aliases
+
+A step can be given an alias by providing an `as` key in the step definition.
+The value of the `as` key is the alias to be used to reference the
+[step's output](#step-outputs).
+
+```yaml
+steps:
+- uses: step-name
+  as: alias
+```
+
+#### Step Outputs
+
+A promotion step may produce output that can be referenced by subsequent steps,
+allowing the output of one step to be used as input to another. The output of a
+step is defined by the step itself and is typically documented in the step's
+reference.
+
+```yaml
+steps:
+- uses: step-name
+  as: alias
+- uses: another-step
+  config:
+    input: ${{ outputs.alias.someOutput }}
+```
+
+:::tip
+If a step does not have an alias, a default alias is generated based on the
+step's position in the `steps` list, starting with `step-0`.
+:::
```

Comment:
Since this was added for internal reasons and I don't think it wise for users to rely on this, I'm questioning whether we're better off not mentioning this detail.

Not a hill I would die on. You decide.

Reply:
Removed it, thanks 🙇